xds/internal/xdsclient/bootstrap/bootstrap.go

```diff
@@ -121,12 +155,13 @@ func (sc *ServerConfig) UnmarshalJSON(data []byte) error {
 	for _, cc := range xs.ChannelCreds {
 		// We stop at the first credential type that we support.
 		sc.CredsType = cc.Type
-		if cc.Type == credsGoogleDefault {
-			sc.Creds = grpc.WithCredentialsBundle(google.NewDefaultCredentials())
-			break
-		} else if cc.Type == credsInsecure {
-			sc.Creds = grpc.WithTransportCredentials(insecure.NewCredentials())
-			break
+		if c := bootstrap.GetCredentials(cc.Type); c != nil {
+			if bundle, err := c.Build(cc.Config); err == nil {
+				sc.Creds = grpc.WithCredentialsBundle(bundle)
+				break
+			} else {
+				logger.Warningf("Failed to build credentials bundle for %q, error: %v", cc.Type, err)
+			}
 		}
```

Comment:
Add `else warning(creds is not registered for %q, cc.Type)`

Reply:
Added.  I thought it might be too spammy as warning considering the logging for it currently does not exist.

Reply:
+1; this seems too spammy to me.  It's definitely not a warning since there's a list of credentials we could fall back to.